lib/Onyx.js

```diff
@@ -34,17 +35,42 @@ let defaultKeyStates = {};
  * @returns {Promise<*>}
  */
 function get(key) {
-    return AsyncStorage.getItem(key)
-        .then(val => JSON.parse(val))
-        .catch(err => logInfo(`Unable to get item from persistent storage. Key: ${key} Error: ${err}`));
+    // When we already have the value in cache - resolve right away
+    if (cache.hasCacheForKey(key)) {
+        return Promise.resolve(cache.getValue(key));
+    }
+
+    /* Otherwise setup a task to retrieve it.
+       This ensures concurrent usages would not start the same task */
+    return cache.resolveTask(`get:${key}`, () => AsyncStorage.getItem(key)
```

Comment:
I guess this method and the description above can have a better name

The idea is the task (2nd parameter) would only be invoked if it needs to be started
E.g. the first call to get `session` would trigger the task and capture it (`get:session`)
While this `session` task is still running other calls arrive, they aren't resolved from cache
as the cache isn't filled yet, but they won't start another task for the same thing
they'll just be hooked to the existing call

I think the method documentation of `cache.resolveTask` describe this well, it's just that
the method name should changed to something like `startTaskIfNecessary` 
And the comment here is a bit confusing
`This ensures concurrent usages would not start the same task`
vs
`This ensures concurrent calls would not start the same task over`